app/components/TodoDetails.tsx

```diff
@@ -0,0 +1,135 @@
+"use client";
+import { TodoCardInfo } from "@/types";
+import {
+  PencilSquareIcon,
+  ClipboardDocumentCheckIcon,
+} from "@heroicons/react/16/solid";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+
+export default function TodoDetails({
+  information,
+  onEdit,
+}: {
+  information: TodoCardInfo;
+  onEdit: (updatedTodo: TodoCardInfo) => void;
+}) {
+  const [isEditing, setIsEditing] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<TodoCardInfo>({
+    defaultValues: {
+      title: information.title,
+      description: information.description,
+      status: information.status,
+      type: information.type,
+    },
+  });
+
+  const onSubmit = (editedInformation: TodoCardInfo) => {
+    onEdit(editedInformation);
+    setIsEditing(false);
+  };
+
+  return (
+    <div className="grid grid-cols-[160px_minmax(400px,600px)_120px] w-fit min-h-[150px] bg-gray-200 rounded-md shadow-md border border-slate-500 overflow-hidden">
+      <div className="flex flex-col justify-center m-3 border-r-2 border-slate-500">
+        <div className="mb-4">
+          <label className="text-slate-500 text-sm block ">Status:</label>
+          {isEditing ? (
```

Comment:
it would be more readable if you create one view for edit mode and one view for readonly mode